server/client.go

```diff
@@ -3597,6 +3613,37 @@ func (c *client) setHeader(key, value string, msg []byte) []byte {
 	return bb.Bytes()
 }
 
+// Will return the value for the header denoted by key or nil if it does not exists.
+// This function ignores errors and tries to achieve speed and no additional allocations.
+func getHeader(key string, hdr []byte) []byte {
+	if len(hdr) == 0 {
+		return nil
+	}
+	index := bytes.Index(hdr, []byte(key))
+	if index < 0 {
+		return nil
+	}
+	index += len(key)
+	if hdr[index] != ':' {
```

Comment:
after moving the index, you need to check that it is not passed the length of `hdr` because otherwise it would panic.

Reply:
Will fix.